app/collectors/data.scala

```diff
@@ -21,31 +21,21 @@ case class JsonDataCollector(origin:JsonOrigin, resource: ResourceType) extends
 case class Data( key:String, values:Seq[Value]) extends IndexedItem {
   def arn: String = s"arn:gu:data:key/$key"
   def callFromArn: (String) => Call = arn => routes.Api.data(arn)
-  def firstMatchingData(stack:Option[String], app:String, stage:String): Option[Value] = {
-    stack.map { s =>
-      values.filter {
-        _.stack.isDefined
-      } find { data =>
-        data.appRegex.findFirstMatchIn(app).isDefined &&
-          data.stageRegex.findFirstMatchIn(stage).isDefined &&
-          data.stackRegex.get.findFirstMatchIn(s).isDefined
-      }
-    } getOrElse {
-      values.filter {
-        _.stack.isEmpty
-      } find {data =>
-        data.appRegex.findFirstMatchIn(app).isDefined && data.stageRegex.findFirstMatchIn(stage).isDefined
-      }
+  def firstMatchingData(stack:String, app:String, stage:String): Option[Value] = {
+    values.find { data =>
+      data.appRegex.findFirstMatchIn(app).isDefined &&
+        data.stageRegex.findFirstMatchIn(stage).isDefined &&
+        data.stackRegex.findFirstMatchIn(stack).isDefined
     }
   }
 }
 
-case class Value( stack: Option[String],
+case class Value( stack: String,
                   app: String,
                  stage: String,
                  value: String,
                  comment: Option[String] ) {
-  lazy val stackRegex = stack.map(s => s"^$s$$".r)
-  lazy val appRegex = ("^%s$" format app).r
-  lazy val stageRegex = ("^%s$" format stage).r
+  lazy val stackRegex = s"^$stack$$".r
+  lazy val appRegex = s"^$app$$".r
```

Comment:
I assume it's intentional the regex of `app` and `stage` have changed to be wrapped in `$`?  Is that for dealing with line breaks in the strings?

Reply:
This hasn't actually changed. `$$` is the literal for `$` when using in the interpolated string.